From d3ef77b3581317056dcd563306fac9c4cb2c0b8d Mon Sep 17 00:00:00 2001
From: Angela Fong <angela.fong@joyent.com>
Date: Thu, 21 Sep 2017 18:01:11 -0700
Subject: [PATCH] ADMINUI-2387 Networking: Read-only operators should not get
 the Delete buttons Reviewed by: Marsell Kukuljevic <marsell@joyent.com>
 Approved by: Marsell Kukuljevic <marsell@joyent.com>

---
 www/js/components/pages/networking/fabric-vlan-list.jsx | 2 +-
 www/js/components/pages/networking/networks-list.jsx    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/www/js/components/pages/networking/fabric-vlan-list.jsx b/www/js/components/pages/networking/fabric-vlan-list.jsx
index 95ce5fb0..800c4453 100644
--- a/www/js/components/pages/networking/fabric-vlan-list.jsx
+++ b/www/js/components/pages/networking/fabric-vlan-list.jsx
@@ -75,7 +75,7 @@ var FabricVlansListItem = React.createClass({
                     <strong>Description</strong>
                     <span className="value">{data.description}</span>
                 </div>
-                {adminui.user.role('operators') || adminui.user.role('readers') ?
+                {adminui.user.role('operators') ?
                     <div className="actions">
                         <a onClick={this.handleEdit} className="edit"><i className="fa fa-pencil"></i> Edit</a>
                         <a onClick={this.deleteVlan} className="delete"><i className="fa fa-trash-o"></i> Delete</a>
diff --git a/www/js/components/pages/networking/networks-list.jsx b/www/js/components/pages/networking/networks-list.jsx
index fe776cc6..fe7c466e 100644
--- a/www/js/components/pages/networking/networks-list.jsx
+++ b/www/js/components/pages/networking/networks-list.jsx
@@ -191,7 +191,7 @@ var NetworksList = React.createClass({
                 <strong>MTU / NIC Tag MTU</strong>
                 <span className="value">{network.mtu} / {this.nicTagsMtu[network.nic_tag]}</span>
             </div>
-            {this.props.showActions && adminui.user.role('operators') || adminui.user.role('readers') ?
+            {this.props.showActions && adminui.user.role('operators') ?
                 (<div className="actions">
                     <a onClick={this.deleteNetwork.bind(this, model)} className="delete"><i className="fa fa-trash-o"></i> Delete</a>
                 </div>)
-- 
2.21.0

